### DISCUSSION TITLE
Update a collection using filter possible ? Js-sdk

### DISCUSSION THREAD

**Discussion by pvbhanuteja** [pocketbase/pocketbase#4070] [Q&A] [answered]
Is there any clean way to do this in a single update using filter?

 ```
 const userProfData = await adminpb
  .collection(Collections.UserProfile)
  .getFirstListItem<UserProfileResponse>(
    pb.filter('user = {:id}', { id: uuid })
  );
  
  const ddd = adminpb.collection(Collections.UserProfile).update<UserProfileResponse>(userProfData.id, {
      sub_id: subs_id,
      next_period: end_date,
      plan: 'paid',
    });
```

Something like 

```
  const ddd = adminpb.collection(Collections.UserProfile).update<UserProfileResponse>(filter,{
      sub_id: subs_id,
      next_period: end_date,
      plan: 'paid',
    });
```

**Comment by ganigeorgiev** [comment]
No. View, update and delete can be performed only via the record id.

**Comment by jonshipman** [comment, marked as answer]
Just write a wrapper function.

```
async function findAndUpdate(collection, filter, data) {
  const record = await adminpb
  .collection(collection)
  .getFirstListItem(filter);
 
  return adminpb.collection(collection).update(record.id, data);
}
```

**Reply by pvbhanuteja** [reply]
Thanks already wrote that. Docs are not great I know its still underactive development love Pockbase for auth handling though.

**Reply by Salman9000** [reply]
Shouldn't there be a function to delete by filter? Almost every query language out there gives this option. It would be great if this is implemented in future as feature.

**Reply by ganigeorgiev** [reply]
@Salman9000 This was discussed several times before. There are no plans for now to allow filter deleting or updating via the Web APIs because there are too many edge cases and will complicate how the API rules are checked. If you really need to delete an item by a filter expression you can create a custom route and perform whatever db query you need.
